### DISCUSSION TITLE
Nouveau format 2

### DISCUSSION THREAD

**Discussion by apprenti-sorcier** [augustinsoulard/KIT_BOTA_QFIELD#5] [General]
Je ne sais pas si c’est l’endroit pour poser la question, mais je n’ai pas compris l’intérêt du changement de format du projet à partir des versions 2.0.
Avantage : il y a moins de fichiers dans le projet.
Inconvénients : la taille du projet a augmenté à cause du fichier DataProjetFlore.gpkg (et donc le temps de chargement). Surtout, on ne peut pas garder de relevés distincts qu’on historise de plusieurs sites qu’on historise sans copier toute cette base qui contient d’autres données. Au moins les taxons restent cohérents avec la base.
Au final, il me semble que les inconvénients dépassent l’avantage.

**Comment by augustinsoulard** [comment]
> Je ne sais pas si c’est l’endroit pour poser la question, mais je n’ai pas compris l’intérêt du changement de format du projet à partir des versions 2.0. Avantage : il y a moins de fichiers dans le projet. Inconvénients : la taille du projet a augmenté à cause du fichier DataProjetFlore.gpkg (et donc le temps de chargement). Surtout, on ne peut pas garder de relevés distincts qu’on historise de plusieurs sites qu’on historise sans copier toute cette base qui contient d’autres données. Au moins les taxons restent cohérents avec la base. Au final, il me semble que les inconvénients dépassent l’avantage.

Il y a pas mal d'avantages aux Geopackages. Notamment le fait de pouvoir y faire des requêtes SQL, d'avoir plus de 254 caractères dans une valeur et même d'avoir des noms de colonnes plus longs. 
Mais finalement, je viens de tout repasser sur le format shapefile car le serveur d'Agence MTDA faisait buguer mes Geopackages...
Mais j'ai trouvé une alternative pour ce qui est des requêtes SQL. Il s'agit des couches virtuels que je connaissais déjà, mais que je n'utilisais pas beaucoup. Finalement, ça me satisfait plutôt bien !

J'ai totalement revu ma méthode de prospection des habitats naturels et l'outil va de pair. Pour la flore, je n'ai pas changé beaucoup de choses.

Augustin


PS : Tu fais bien de mettre tes remarques ici et je t'en remercie. J'avais vu ton post depuis un moment, mais j'attendais de tester vraiment avant de te répondre ;)

**Comment by apprenti-sorcier** [comment]
Merci pour les explications.
Est‐ce à cause des couches virtuelles que la taille du fichier DataProjetFlore.gpkg est 6 fois plus grande que dans la v. 1.3 par exemple ? Ou il y a des données en double ?

(Par ailleurs la couche « Enjeu floristique PACA » fait référence à un fichier local, indisponible pour les autres utilisateurs.
Et dans le plugin QFieldSync, la fonction « Paquet pour QField » propose par défaut un dossier d’export dans des dossiers ctera faisant références à certaines études…)

**Reply by augustinsoulard** [reply]
DataProjetFlore.gpkg est plus lourd car j'ai rajouté tous les statuts de la flore de PACA dedans. 
Enjeu floristique PACA n'est pas vraiment une couche, c'est justement une requête SQL sur Flore pour avoir les enjeux des espèces qu'on vient de pointer directement sur le terrain. C'est très utile en bureau d'étude surtout pour les plus débutants qui ne connaissent pas encore tous les enjeux. A terme je voudrais faire ça pour plus de régions avec une requête qui compare la position du point avec les limites régionales et ira chercher les bons statuts. Ca va forcément alourdir le projet par contre :/ Il faudra que je réfléchisse à comment limiter ça.

**Reply by augustinsoulard** [reply]
Concernant QfieldSync c'est l'historique du projet Qgis qui fait ça. J'ai changé le projet en transférant toutes les couches sur un autre projet. J'espère que ça fonctionnera pour ne plus avoir tout ces restes de l'époque où j'étais à la réunion et j'avais déjà commencé un premier projet.

Encore merci de tes retours !
